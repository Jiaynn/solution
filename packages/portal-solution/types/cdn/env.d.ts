/**
 * @file env config
 * @author linchen <linchen@qiniu.com>
 */

declare interface OEMConfig {
  logo: string
  vendor: number
  favicon: string
  title: string
  copyright: string
  lang: 'en' | 'cn'
  domainHosting: boolean                  // 域名托管功能
  financial: boolean                      // 财务中心
  hideLoginTitle: boolean                 // 隐藏登录页面的 title
  hideCreateDomain: boolean               // 隐藏创建域名的操作入口
  hideUpdateDomain: boolean               // 隐藏更新域名的操作入口
  hideSubAccountDomainListCname: boolean  // 对子用户隐藏域名列表页的 cname
  hideSubAccountDomainDetailCname: boolean // 对子用户隐藏域名配置详情页的 cname
  hideSubAccountDomainOperations: boolean // 对子用户隐藏域名的操作 (只保留统计操作)
  hideSubAccountCreateDomain: boolean     // 对子用户隐藏创建域名的操作
  hideSubAccountGeoForeign: boolean       // 对子用户禁用海外覆盖范围
  hideSubAccountGeoGlobal: boolean        // 对子用户禁用全球覆盖范围
  hideSubAccountStatsUsagePeakAverage: boolean    // 对子用户隐藏平均峰值带宽
  hideSubAccountStatsUsagePeak95Average: boolean  // 对子用户隐藏平均 95 峰值带宽
  statisticsDateRangeDays: number // 数据统计默认时间范围, 最近的天数
  statisticsFreq: string // 数据统计粒度 '1day' | '1hour' | '5min'
}

declare let BUILD_TARGET: 'qiniu' | 'oem'
declare let OEM_CONFIG: OEMConfig
declare let APP_VERSION: string
