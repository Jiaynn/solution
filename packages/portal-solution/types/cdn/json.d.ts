declare let jsonExport: any

declare module '*.json' {
  export default jsonExport
}
