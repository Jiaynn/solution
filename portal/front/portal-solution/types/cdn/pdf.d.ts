declare let pdfExport: string

declare module '*.pdf' {
  export = pdfExport
}
