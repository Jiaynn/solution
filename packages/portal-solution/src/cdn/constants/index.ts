// 启用动态加速工单地址
export const enableDynamicURL = 'https://support.qiniu.com/tickets/new/form?category=%E5%85%B6%E4%BB%96%E7%B1%BB%E5%92%A8%E8%AF%A2&space=%E8%9E%8D%E5%90%88%20CDN'

// 证书配置提交工单 URL
export const certConfigHelpURL = 'https://support.qiniu.com/tickets/new/form?category=%E8%AF%81%E4%B9%A6%E9%97%AE%E9%A2%98&space=CDN'
