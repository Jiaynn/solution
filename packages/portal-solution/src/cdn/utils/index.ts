import { values } from 'lodash'
import { ReactNode } from 'react'
import { makeCancelled, UnknownException } from 'qn-fe-core/exception'

export function isPromise(obj: any): obj is Promise<any> {
  if (!obj) {
    return false
  }
  if (typeof obj === 'object' || typeof obj === 'function') {
    if (typeof obj.then === 'function') {
      return true
    }
  }
  return false
}

export interface IAntdStyleMethodOptions {
  title?: ReactNode
  content?: ReactNode
  onOk?: (...args: any[]) => void
  onCancel?: (...args: any[]) => void
}

export type AntdStyleMethod<T extends IAntdStyleMethodOptions> = (options: T) => void

export type PromiseStyleMethod<T, P> = (options: T) => Promise<P>

// 把 antd 中如 Modal.confirm 风格（onCancel, onOk）的接口转换成 promise 接口
export function antdToPromise<
  T extends IAntdStyleMethodOptions,
  M extends AntdStyleMethod<T>,
  P
>(method: M): PromiseStyleMethod<T, P> {
  return (options: T) => new Promise((resolve, reject) => {
    const onCancel = (...args: any[]) => {
      if (options.onCancel) {
        options.onCancel(...args)
      }
      reject(makeCancelled())
    }
    const onOk = (...args: any[]) => {
      if (options.onOk) {
        options.onOk(...args)
      }
      resolve(args[0])
    }
    method({
      ...(options as any),
      onCancel,
      onOk
    })
  })
}

function getLexicValue(char: string) {
  return char ? char.charCodeAt(0) : -1
}

export function lexicCompare(strA: string, strB: string) {
  for (let i = 0, len = Math.max(strA.length, strB.length); i < len; i++) {
    const vA = getLexicValue(strA.charAt(i))
    const vB = getLexicValue(strB.charAt(i))
    if (vA !== vB) {
      return vA - vB
    }
  }
  return 0
}

export function valuesOfEnum<T extends object>(target: T): Array<T[keyof T]> {
  const fakeKeys = values(target).filter(value => typeof value !== 'string').map(String)
  const realKeys = Object.keys(target).filter(key => fakeKeys.indexOf(key) === -1)

  return (realKeys as Array<keyof T>).map(key => target[key])
}

export function assertUnreachable(): never
export function assertUnreachable(v: never, message?: string): never
export function assertUnreachable(v?: never, message = "Didn't expect to get here") {
  throw new UnknownException(message + v)
}

/**
 * 过滤数组的 “空” 值，其中空值指的是：
 * - null
 * - undefined
 */
export function nonEmptyArray<T>(data: Array<T | undefined | null>): T[] {
  return data.filter((it): it is T => it != null)
}

export function booleanPredicate<T>(value: T): value is Exclude<T, false | null | undefined | '' | 0> {
  return Boolean(value)
}

export function assert(condition: any, msg?: string): asserts condition {
  if (!condition) {
    throw Error(msg)
  }
}
