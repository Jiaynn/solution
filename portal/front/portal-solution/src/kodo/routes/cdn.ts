/**
 * @file CDN route TODO: 站内外链与内链分开
 * @author yinxulai <me@yinxulai.com>
 */

// CDN 概览
export function getCDNOverviewPath() {
  return '/cdn/overview'
}

// CDN 业务的域名创建
export function getCDNCreateDomainPath(bucketName: string) {
  const query = bucketName ? `?bucket=${encodeURIComponent(bucketName)}&fixBucket` : ''
  return `/cdn/domain/create${query}`
}

// CDN 业务的域名列表
export function getCDNDomainListPath() {
  return '/cdn/domain'
}

export function getCDNDomainDetailPath(domain: string) {
  return `/cdn/domain/${domain}`
}

export function getCDNCreateBucketDomainPath(bucketName: string) {
  return `/cdn/domain/create?bucket=${bucketName}&fixBucket`
}

export function getCDNRefreshPath() {
  return '/cdn/refresh-prefetch'
}

export function getCDNRefreshLogsPath() {
  return '/cdn/refresh-prefetch?tab=showLogs'
}
