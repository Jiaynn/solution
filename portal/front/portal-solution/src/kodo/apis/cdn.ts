/**
 * @file config api
 * @description region 相关的接口
 * @author lizhifeng <lizhifeng@qiniu.com>
 */

import autobind from 'autobind-decorator'
import { injectable } from 'qn-fe-core/di'
import { ApiException } from 'qn-fe-core/client'

import { ProxyClientV2 } from 'portal-base/common/apis/proxy'

import { proxy } from 'kodo/constants/apis'

interface CdnBaseResponse {
  code: number
  error: string
  [k: string]: any
}

export interface RefreshCdnSurplus extends CdnBaseResponse {
  urlQuotaDay: number // url 刷新当日限额
  urlSurplusDay: number // url 刷新当日余量
  dirQuotaDay: number // 目录刷新当日限额
  dirSurplusDay: number // 目录刷新当日余量
  quotaDay: number // 预取当日限额
  surplusDay: number // 预取当日剩余量
}

interface RefreshCdnOptions {
  product: 'cdn'
  urls: string[]
}

export class CdnApiException extends ApiException {
  constructor(
    public code: number,
    public payload: CdnBaseResponse,
    message?: string,
    cause?: unknown
  ) {
    super(
      'RefreshPrefetchApiException',
      code,
      message,
      undefined,
      cause
    )
  }
}

const messageMap = new Map<number, string>([
  [400031, '无效的 URL'],
  [400034, '刷新域名个数超过限制'],
  [400037, 'URL 已在队列中，无需重复提交'],
  [400039, '请求次数过多']
])

function produceResult<T extends CdnBaseResponse>(payload: T) {
  if (payload && typeof payload === 'object' && 'code' in payload && payload.code !== 200) {
    const message = messageMap.get(payload.code) || payload.error
    throw new CdnApiException(payload.code, payload, message)
  }

  return payload
}

@autobind
@injectable()
export class CdnApis {
  constructor(
    private proxyClient: ProxyClientV2
  ) {}

  getRefreshCdnSurplus(): Promise<RefreshCdnSurplus> {
    return this.proxyClient.get<RefreshCdnSurplus>(proxy.refreshCdnSurplus, {})
      .then(produceResult)
  }

  async refreshCdn(params: RefreshCdnOptions): Promise<void> {
    await this.proxyClient.post<CdnBaseResponse>(proxy.refreshCdn, params)
      .then(produceResult)
  }
}
