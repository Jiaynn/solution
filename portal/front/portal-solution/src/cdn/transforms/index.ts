import { QueryParams } from 'qn-fe-core/utils'

export function atoi(numberText: string, valueForInvalidInput = 0): number {
  const parsed = parseInt(numberText, 10)
  return Number.isNaN(parsed) ? valueForInvalidInput : parsed
}

export function atof(numberText: string, valueForInvalidInput = 0): number {
  const parsed = parseFloat(numberText)
  return Number.isNaN(parsed) ? valueForInvalidInput : parsed
}

export function getIntValue(e: React.FormEvent<any>, valueForInvalidInput = 0) {
  return atoi((e.target as any).value, valueForInvalidInput)
}

export function getFloatValue(e: React.FormEvent<any>, valueForInvalidInput = 0) {
  return atof((e.target as any).value, valueForInvalidInput)
}

export function trimAndFilter(lines: string[]): string[] {
  return (lines || []).map(
    line => line.trim()
  ).filter(
    line => !!line
  )
}

export function splitLines(linesStr: string | null | undefined): string[] {
  if (!linesStr) {
    return []
  }
  return linesStr.split('\n')
}

export function filterQuery(query?: QueryParams) {
  if (!query) {
    return
  }
  return Object.keys(query).reduce(
    (filtered, key) => (
      (query || {})[key]
      ? { ...filtered, [key]: (query || {})[key] }
      : filtered
    ), {} as QueryParams
  )
}

export function humanizeOnoff(isOn: boolean) {
  return isOn ? '开启' : '关闭'
}
