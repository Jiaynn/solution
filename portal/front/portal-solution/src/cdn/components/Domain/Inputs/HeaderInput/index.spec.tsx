/*
 * @file cases for domain http header input form
 * @author zhuhao <zhuhao@qiniu.com>
 */

import React from "react";

import { RendererUtils as Renderer } from "cdn/test";

import mockDomainDetail from "cdn/test/domain-detail-mock";

import {
  ResponseHeaderControlKey,
  ResponseHeaderControlOp,
} from "cdn/constants/domain";

import HttpHeaderInput, {
  createState,
} from "cdn/components/Domain/Inputs/HeaderInput";

const domain = mockDomainDetail();

it("renders correctly", () => {
  const state = createState([
    {
      key: ResponseHeaderControlKey.ContentLanguage,
      value: "zh-CN",
      op: ResponseHeaderControlOp.Set,
    },
  ]);
  const tree = new Renderer()
    .createWithAct(<HttpHeaderInput domain={domain} state={state} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
