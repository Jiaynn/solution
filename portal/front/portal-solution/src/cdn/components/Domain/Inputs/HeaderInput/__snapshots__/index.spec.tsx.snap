// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="domain-response-header-input-wrapper"
>
  <table
    className="respheader-table"
  >
    <thead>
      <tr>
        <th>
          HTTP 响应头
        </th>
        <th>
          响应头操作
        </th>
        <th>
          值
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr
        className="sources-table-line"
      >
        <td
          className="sources-table-grid-key"
        >
          <div
            className="ant-select ant-select-enabled"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseOut={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            <div
              aria-autocomplete="list"
              aria-controls="test-uuid"
              aria-expanded={false}
              aria-haspopup="true"
              className="ant-select-selection
            ant-select-selection--single"
              onKeyDown={[Function]}
              role="combobox"
              tabIndex={0}
            >
              <div
                className="ant-select-selection__rendered"
              >
                <div
                  className="ant-select-selection__placeholder"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  style={
                    Object {
                      "WebkitUserSelect": "none",
                      "display": "none",
                      "userSelect": "none",
                    }
                  }
                  unselectable="on"
                >
                  请选择 HTTP 响应头
                </div>
                <div
                  className="ant-select-selection-selected-value"
                  style={
                    Object {
                      "display": "block",
                      "opacity": 1,
                    }
                  }
                  title="Content-Language"
                >
                  Content-Language
                </div>
              </div>
              <span
                className="ant-select-arrow"
                onClick={[Function]}
                style={
                  Object {
                    "WebkitUserSelect": "none",
                    "userSelect": "none",
                  }
                }
                unselectable="on"
              >
                <i
                  aria-label="icon: down"
                  className="anticon anticon-down ant-select-arrow-icon"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </i>
              </span>
            </div>
          </div>
        </td>
        <td
          className="sources-table-grid-op"
        >
          <div
            className="ant-select ant-select-enabled"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseOut={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            <div
              aria-autocomplete="list"
              aria-controls="test-uuid"
              aria-expanded={false}
              aria-haspopup="true"
              className="ant-select-selection
            ant-select-selection--single"
              onKeyDown={[Function]}
              role="combobox"
              tabIndex={0}
            >
              <div
                className="ant-select-selection__rendered"
              >
                <div
                  className="ant-select-selection__placeholder"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  style={
                    Object {
                      "WebkitUserSelect": "none",
                      "display": "none",
                      "userSelect": "none",
                    }
                  }
                  unselectable="on"
                >
                  请选择操作
                </div>
                <div
                  className="ant-select-selection-selected-value"
                  style={
                    Object {
                      "display": "block",
                      "opacity": 1,
                    }
                  }
                  title="设置"
                >
                  设置
                </div>
              </div>
              <span
                className="ant-select-arrow"
                onClick={[Function]}
                style={
                  Object {
                    "WebkitUserSelect": "none",
                    "userSelect": "none",
                  }
                }
                unselectable="on"
              >
                <i
                  aria-label="icon: down"
                  className="anticon anticon-down ant-select-arrow-icon"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </i>
              </span>
            </div>
          </div>
        </td>
        <td
          className="sources-table-grid-value"
        >
          <input
            className="ant-input"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="请输入对应值, 区分大小写"
            type="text"
            value="zh-CN"
          />
        </td>
        <td
          className="sources-table-grid-operation"
        >
          <i
            aria-label="icon: minus-circle"
            className="anticon anticon-minus-circle icon-remove"
            onClick={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="minus-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M696 480H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h368c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
              />
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
            </svg>
          </i>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
  <button
    className="ant-btn add-tag ant-btn-dashed"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <i
      aria-label="icon: plus"
      className="anticon anticon-plus"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="plus"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
        />
        <path
          d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
        />
      </svg>
    </i>
    <span>
      添加
    </span>
  </button>
</div>
`;
